fix(ngxs): upgrade to NGXS 3.8

gromon42 opened the pull request:
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [x] The commit message follows our guidelines: https://github.com/ngxs/store/blob/master/CONTRIBUTING.md#commit
-   [x] Tests for the changes have been added (for bug fixes / features)
-   [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

The compilation is broken on Angular 15 since the ngxs does not has the proper version.

Link to the issue ie [here](https://github.com/Angular-RU/angular-ru-sdk/issues/1459)

## What is the new behavior?

It no longer spawns error like : 

![Screenshot from 2023-04-05 16-33-48](https://user-images.githubusercontent.com/16048524/230113873-b3e4e0d1-904c-4498-902b-71fb80e41fd5.png)


## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

gromon42 commented:
> <img alt="image" width="1320" src="https://user-images.githubusercontent.com/12021443/230114802-afc2ff81-6d12-4634-8174-a3167c99793b.png">

Hmmm i will look more into it, will keep you posted and update the issue accordingly

gromon42 commented:
I've replaced the broken imports !

splincode commented:
`$ nx build ngxs`?

splincode commented:
<img width="829" alt="image" src="https://user-images.githubusercontent.com/12021443/230135421-2ae06f34-3f1a-4437-a5a0-13e9d0c35ee2.png">

gromon42 commented:
The build is now passing, i have no clue how to fix the tests though, do you have any idea why there is an error triggered ? 

I have not touched anything regarding this part of the code.

gromon42 commented:
> <img alt="image" width="829" src="https://user-images.githubusercontent.com/12021443/230135421-2ae06f34-3f1a-4437-a5a0-13e9d0c35ee2.png">

Fixed, apparently the error returned by NGXS has changed, I fixed it accordingly.

gromon42 commented:
All tests are now passing !

splincode commented on libs/ngxs/internals/src/services/ngxs-data-injector.service.ts:
```diff
@@ -1,6 +1,6 @@
 import { Inject, Injectable, Injector, NgZone } from '@angular/core';
 import { Store } from '@ngxs/store';
-import { NGXS_STATE_CONTEXT_FACTORY, NGXS_STATE_FACTORY } from '@ngxs/store/internals';
+import { ɵNGXS_STATE_CONTEXT_FACTORY, ɵNGXS_STATE_FACTORY } from '@ngxs/store/internals';
```
```ts
import { ɵNGXS_STATE_CONTEXT_FACTORY as NGXS_STATE_CONTEXT_FACTORY, ɵNGXS_STATE_FACTORY as NGXS_STATE_FACTORY }
```

splincode commented on libs/ngxs/internals/src/services/ngxs-data-injector.service.ts:
```diff
@@ -15,8 +15,8 @@ export class NgxsDataInjector {
         injector: Injector,
-        @Inject(NGXS_STATE_FACTORY) stateFactory: unknown | any,
-        @Inject(NGXS_STATE_CONTEXT_FACTORY) stateContextFactory: unknown | any
+        @Inject(ɵNGXS_STATE_FACTORY) stateFactory: unknown | any,
```
revert

splincode commented on package.json:
```diff
@@ -59,8 +59,8 @@
         "@ngx-translate/http-loader": "7.0.0",
-        "@ngxs/logger-plugin": "3.7.3",
-        "@ngxs/store": "3.7.3",
+        "@ngxs/logger-plugin": "^3.8.0",
+        "@ngxs/store": "^3.8.0",
```
please, pin versions 

```json
"@ngxs/logger-plugin": "3.8.0",
"@ngxs/store": "3.8.0",
```

and rebuild lock file

gromon42 replied:
done
